concourse/pipelines/create-bosh-cloudfoundry.yml

```diff
@@ -483,6 +486,59 @@ jobs:
           params:
             file: generated-cf-secrets/cf-secrets.yml
 
+
+  - name: pre-deploy
+    plan:
+      - aggregate:
+          - get: pipeline-trigger
+            passed: ['generate-secrets']
+            trigger: true
+          - get: paas-cf
+            passed: ['generate-secrets']
+          - get: cf-secrets
+            passed: ['generate-secrets']
+          - get: bosh-CA
+            passed: ['generate-secrets']
+      - try:
+          task: wait-for-availability-test
+          config:
+            platform: linux
+            inputs:
+              - name: bosh-CA
+              - name: paas-cf
+              - name: cf-secrets
+              - name: pipeline-trigger
+            params:
+              SYSTEM_DNS_ZONE_NAME: {{system_dns_zone_name}}
+              CF_ADMIN: admin
+            image: docker:///governmentpaas/cf-cli
+            run:
+              path: sh
+              args:
+                - -e
+                - -u
+                - -c
+                - |
+                  ./paas-cf/concourse/scripts/import_bosh_ca.sh
+
+                  CF_PASS=$(awk '/uaa_admin_password/ {print $2}' cf-secrets/cf-secrets.yml)
+                  API_ENDPOINT="https://api.${SYSTEM_DNS_ZONE_NAME}"
+                  PIPELINE_TRIGGER_VERSION=$(cat pipeline-trigger/number)
+
+                  echo | cf login -a ${API_ENDPOINT} -u ${CF_ADMIN} -p ${CF_PASS} > /dev/null
+                  cf target -o testers -s healthcheck > /dev/null
+
+                  echo "Waiting two minutes for availability test to start:"
+                  for i in $(seq 24); do
+                    if cf logs healthcheck --recent | grep -q "availability-test=${PIPELINE_TRIGGER_VERSION}"; then
+                      echo "Request detected"
+                      break
```

Comment:
how about exit 0 here and exit 1 after done? This way pipeline can proceed only if it is monitored...